hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java

```diff
@@ -115,11 +116,17 @@ protected void execute(OzoneClient client, OzoneAddress address)
         ContainerChunkInfo containerChunkInfo = new ContainerChunkInfo();
         long containerId = keyLocation.getContainerID();
         chunkPaths.clear();
-        Pipeline pipeline = keyLocation.getPipeline();
-        if (pipeline.getType() != HddsProtos.ReplicationType.STAND_ALONE) {
-          pipeline = Pipeline.newBuilder(pipeline)
+        Pipeline keyPipeline = keyLocation.getPipeline();
+        boolean isECKey =
+            keyPipeline.getReplicationConfig().getReplicationType() ==
+                HddsProtos.ReplicationType.EC;
+        Pipeline pipeline;
+        if (keyPipeline.getType() != HddsProtos.ReplicationType.STAND_ALONE) {
+          pipeline = Pipeline.newBuilder(keyPipeline)
               .setReplicationConfig(StandaloneReplicationConfig
                   .getInstance(ONE)).build();
+        } else {
+          pipeline = keyPipeline;
```

Comment:
I see you are using the same pipeline as found in the keyLocation to read the data. This was a problem in RATIS pipeline fixed in HDDS-8546 and might occur here too if the pipeline is closed.

Reply:
Previous logic:

```
pipeline = keyPipeline
if (not standalone) {
  pipeline = ...
}
```

New logic:

```
if (not standalone) {
  pipeline = ...
} else {
  pipeline = keyPipeline
}
```

I think the result is the same in both cases.

Reply:
Thank you @sadanand48  for pointing this out. I checked the code again and I don't see how my changes could be affected by the JIRA you mentioned. 
What I did - I just reordered the `pipeline` variable initialization and extracted all `keyLocation.getPipeline()` usages to a separate variable. Also I checked the case from HDDS-8546 and it appears to work.